lib/Runtime/Library/JavascriptProxy.cpp

```diff
@@ -919,11 +912,38 @@ namespace Js
         Var trapResult = getEnumeratorMethod->CallFunction(arguments);
         threadContext->SetImplicitCallFlags((Js::ImplicitCallFlags)(saveImplicitCallFlags | ImplicitCall_Accessor));
 
-        if (!JavascriptOperators::IsObject(trapResult))
+        if (!JavascriptOperators::IsArray(trapResult))
         {
-            JavascriptError::ThrowTypeError(scriptContext, JSERR_InconsistentTrapResult, _u("enumerate"));
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_InconsistentTrapResult, _u("ownKeys"));
         }
-        *enumerator = IteratorObjectEnumerator::Create(scriptContext, trapResult);
+
+        JavascriptArray* arrTrapResult = JavascriptArray::FromVar(trapResult);        
+        uint32 len = arrTrapResult->GetLength();
```

Comment:
need a generic method to iterate the array (or array like) elements? do we have such method?

---
In reply to: [63966947](https://github.com/Microsoft/ChakraCore/pull/976#discussion_r63966947) [](ancestors = 63966947,63961681)